src/components/rubric-criteria-group.js

```diff
@@ -0,0 +1,1096 @@
+			return null;
+		}
+
+		return this.localize('addFeedbackFor', 'criterionName', criterion.name);
+	}
+	_localizeLevelOutOf(type, points) {
+		if (points === undefined || points === null) {
+			return;
+		}
+		return this.localize(type, 'number', points.toString());
+	}
+	_lookupMap(entity, map) {
+		if (!entity || !map) {
+			return null;
+		}
+
+		const selfLink = this._getSelfLink(entity);
+		return selfLink && selfLink !== '' ? map[selfLink] : null;
+	}
+	_onEntityChanged(entity) {
+		this._levelsHref = entity.levelsLink;
+		this._criteriaCollectionHref = entity.criteriaLink;
+		// The first time that the entity loads, we send out an Open event
+		if (!this._loaded) {
+			const entityId = this._getSelfLink(entity);
+			this.logViewRubricEvent({ id: entityId });
+			this._loaded = true;
+		}
+	}
+	_queueChange(entityGetter) {
+		// TODO, when updating the parent component change it to use a more specialized entity class
+		const entity = entityGetter();
+		const criterionHref = entity.getLinkByRel('https://rubrics.api.brightspace.com/rels/criterion')?.href;
+		const levelHref = entity.getLinkByRel('https://rubrics.api.brightspace.com/rels/level')?.href;
+		if (this.CriterionAssessmentHelper.isOptimisticallySelected(entity, this._optimisticSelections)) {
+			this._dispatcher.selectLevel(criterionHref, null);
+			this._optimisticSelections.set(criterionHref, null);
+			this._optimisticSelectionsVersion++;
+		} else {
+			this._dispatcher.selectLevel(criterionHref, levelHref);
+			this._optimisticSelections.set(criterionHref, levelHref);
+			this._optimisticSelectionsVersion++;
+		}
+		this._dispatcher.selectLevel(criterionHref, levelHref);
+
+		const uuid = this.getUUID();
+		this.perfMark(`criterionCellTappedStart-${uuid}`);
+		this.logCriterionCellTappedAction(`criterionCellTappedStart-${uuid}`, `criterionCellTappedEnd-${uuid}`);
+		this.CriterionCellAssessmentHelper.selectAsync(entityGetter).then(() => {
+			this.logCriterionCellTappedAction(`criterionCellTappedStart-${uuid}`, `criterionCellTappedEnd-${uuid}`);
+		}).catch(err => {
+			this.optimisticUpdateDispatcher.rollback(criterionHref);
+			this._optimisticSelections.delete(criterionHref);
+			this._optimisticSelectionsVersion++;
+			throw err;
+		});
+	}
+	_queueRadio(radio) {
+		const href = radio.getAttribute('data-href');
+		const entity = this.cellAssessmentMap[href];
+		const criterionHref = entity?.getLinkByRel('https://rubrics.api.brightspace.com/rels/criterion')?.href;
+		if (criterionHref) {
+			this._queueChange((() => this.cellAssessmentMap[href]).bind(this));
+		}
+	}
+	_renderCriterion(criterion, criterionNum) {
+
+		let rowHeader = null;
+		if (this._showRowHeaders(this.rubricType)) {
+
+			let competenciesIcon = null;
+			if (this._showCompetencies(this.criterionResultMap, criterion, this.readOnly)) {
+				competenciesIcon = html`
+					<d2l-rubric-competencies-icon
+						competency-names="${this._getCriterionCompetencies(this.criterionResultMap, criterion)}"
```

Comment:
```suggestion
						.competencyNames="${this._getCriterionCompetencies(this.criterionResultMap, criterion)}"
```
This'll propagate as `[object Object]` otherwise